src/pfe/file-watcher/server/src/utils/workspaceSettings.ts

```diff
@@ -98,6 +98,11 @@ export async function readWorkspaceSettings(): Promise<IWorkspaceSettingsSuccess
 }
 
 export async function writeWorkspaceSettings(address: string, namespace: string): Promise<IWorkspaceSettingsSuccess | IWorkspaceSettingsFailure> {
+    // Clean up namespace if user enters address/namespace in the UI
+    if (namespace.startsWith(address.replace(/\/\s*$/, "") + "/")) {
+        namespace = namespace.replace(address.replace(/\/\s*$/, "") + "/", "");
+    }
+
```

Comment:
I don't think this is the right fix. The address and namespace parameters should be validated and sanitized when received in the REST API here:
https://github.com/eclipse/codewind/blob/2a860de3981d06b81177d7fa93d903b470ac55a1/src/pfe/portal/routes/imagePushRegistry.route.js#L51

Looking at that code the namespace is altered if the settings are used in `writeWorkspaceSettings` but not in the call to `testImagePushRegistry`. I think they should be validated and cleaned up early as part of the REST API before either of those calls are made. That way if there's a problem the REST API can return an appropriate error straight away without having to return it all the way back up the call stack and the same fixes and validation will apply to the parameters passed to `writeWorkspaceSettings` and `testImagePushRegistry`.

Reply:
makes sense, let me see how i can clean it up